### PR TITLE
Huge Pile of Uplink Shit

### PR DESCRIPTION
**Pull request by RandomColonist**
- Multiplied all prices in uplink by ten, as well as basic amount (how much you usually get to spend). This will allow more precise prices adjustment. I, however, not touched some prices at all, and changed others as follow.
- ^But basic amount is 240, so it was multiplied by 9 for reasons. We can divide more things by 2 then by 5.
- Remade Ammo category in uplink. Added regular ammo for 2/3 of high-velocity price of the same type. Rebalanced prices a bit too. Remade names and order.
- Remade Grenades and Throwable Items category in uplink. You can now buy grenades as single or in bundles. Bundle contains six grenades for the price of five, so it's more profitable to buy them in bulk. Rebalanced prices in this section also. Added missing grenades to it. Made cheap grenades like flashbangs cost less and robust grenades like blast cost way more.
- For the sake of consistency, each box of grenades now contains six of them.
- New category, Apparel and Armor at uplink. Contains, you know, Apparel and Armor. Tactical vest, tactical helmet, boots, all useful types of pouches, belts and holster.
- No-slip shoes are chameleons now, retaining theis no-slipness and silence. You can evade slipping with style now. Can't hold knife even if you will make them look like jackboots.
- Added swat boots to uplink, reduced their siemens coefficient. They can hold a knife now. Renamed them to tactical boots.
- Tan webbed vest are tactical vest now, added matching helmet for it. Helmet has slightly more ballistic defence to discourage people to aim at head. Stats pretty much like IH armor, but vest has webbing and helmet covers more and has slightly more ballistic damage, but prevents gear manipulation under it. Original sprites by WalterJe, but slim, female, female slim, fat sprites by Shazbot. (slim helmet replaced with regular one)
serbguys
- Renamed old tactical armor and helmet to NT armor and NT helmet, added them to junk spawner, NT armor slowdown is 0.5 instead of 1.
- Added new vest, helmet junk spawners, so you can find these in junk now.
- Voice changing mask are chameleon mask now, but retains it possibility to change voice. You can make it look like a cig and walk with voice-changing mask unnoticed.
- Added new stealth weapon - disguised syringe gun. Just a regular syringe gun, but a bit weaker and looks exactly like regular unlit cigarette (you can't put it in face slot tho, sadly, but can carry behind the ear). Comes in a kit with four syringes and cartridges. Toxins are not included. Can be carried in regular cig packet.
- Added (stolen from bay and i don't even regret it, Eris deserves it much more) new Badass weapon to an uplink - money cannon. Shoots money, obviously. Can dispence (read - shoot) money at different amounts. Guild are being annoying? Pay them 1000 in 10 thallers bills right in their face. If you emag it - money, beginning at 100-bills would hurt. 2 millions gibs unprotected head. Try to suicide with it for something funny. It's available in "Badassery" section of an uplink, as well as in junk piles.
- Remade Medical section of an uplink. Added (O-) bloodpack, as well as colored injectors with various chems. Sprites by Undud. Also split combat medkit into two medkits, drug firstaid kit and combat firstaid kit.
- Rotated injector sprites upwards for nice look in medkits.
- Remade old combat medkit into drug kit. Contains same pillbottles, but splints are replaced with cheap, non-refillable hyperspray pre-filled with tricordrazine. Modified sprites of original hypo to make it look like cheap shit for this one.
- Made new combat medkit which is now actually useful for tending combat wounds. Includes two advanced traumakit, one advanced burn kit, two bicaridine,  two dermaline, one dexaline plus, one inaprovaline autoinjectors as well as combat hypospray filled with synaptizine, hyperizine and oxycodone. Non-reffilable. Sorry, couldn't implement clotting agent now.
- Removed retarded things from Serbian goals like protect.
- Both bioweapon sample and ballistic drop pod now cost 240 crystals. Bioweapon sample is literally a meme of blob-in-a-can, now serbs can buy only two of those and won't get any other equipment from an uplink. Ballistic drop pod can crush literally anything including substation which is nearly impossible to repair and affects a huge area with an explosion, as well as delievering some supplies.
- Made access downloading considerably easier with access decription program, which can be downloaded to emagged modular console. Now it has lower skill requirements, so even assistants should use it somewhat effective.
- Fixed nasty bug with inability to click shortly after unbuckling.

Might forgot something, feel free to correct me in comments.

### CODE REVIEW

**Comment by RandomColonist** [comment]
Fixed, ready for a review.

**Comment by driecg36** [comment]
I actually like "protect" objectives for serbs. Gives them an incentive to actually pick their targets and not slaughter everyone indiscriminately, which could lead to more RP.

**Comment by zamolxius** [comment]
you're misusing the term "RP", as even the role of a indiscriminate slaughterer is fitting for a serb that has it as an objective, or has objectives requiring it. Even if not, I don't think a space serbian mercenary deathsquad has scruples about unfortunate casualties. 
"RP" is overused and a buzzword, a better term is interaction, in your case non hostile interaction.

**Comment by driecg36** [comment]
I think RP is just a fill-in term for any sort of interaction that involves meaningful, in-character interaction that isn't combat. Anyway, whatever the word is for it is, I still think it's worth leaving in so that serbians have motivation to interact with the crew beyond indiscriminate slaughter. However they justify this (this person has some sort of connection to the people who hired them, they have a sense of honor, etc) is up to them.

**Comment by RandomColonist** [comment]
> I think RP is just a fill-in term for any sort of interaction that involves meaningful, in-character interaction that isn't combat. Anyway, whatever the word is for it is, I still think it's worth leaving in so that serbians have motivation to interact with the crew beyond indiscriminate slaughter. However they justify this (this person has some sort of connection to the people who hired them, they have a sense of honor, etc) is up to them.

Protip: we are de-facto Low-RP, thus Serbs are just valids. If you want come up with gimmick - very well, disregard your goals and do it yourself, literally nothing stops you, but game itself shouldn't suggest nukeops to not kill people. If you need motivation like greentext for not murderboning (playing role that was literally made for it), well, it's your personal issue. I will not change it in my PR.

**Comment by RandomColonist** [comment]
Man, just fucking, like, turn on your brain and think - you are actually defending objective which supposed to restrict fucking serbian ops armed with only a lot of lethals from killing someone. Like holy fuck, last time i was supposed to defend a fucking operative. If you with to continue this retardness - respond on discord.

**Comment by RandomColonist** [comment]
Ready to review and merge.

**Comment by SpadesSlick** [comment]
Okay, I have to stop you right here.

We allowed you to group up thing up to 10 changes in one PR before, but this is ridiculous. There are 21 parts of this PR, each of them should be reviewed separately, and it slowing down this whole PR.

It's painful to review this way, needlessly large and can contain things that both can be merged right away, and should be discussed more deeply. Split them to separate PRs so we will be able to do this normally. It's not that hard.

**Comment by SpadesSlick** [comment]
ed47f358860bfe5c5481fc4d9096fd1e
okay fine

Splitting that uplink datum will be to much pain for you.

But if you, or someone else, bring something like this again, I swear to god I will do things to you so horrible, that github mods would ban me (and this whole repo with me) if I post it right here.

File: code/modules/clothing/head/helmet.dm [hunk 1]
```diff
@@ -58,16 +58,24 @@
...
 	siemens_coefficient = 0.7
 	price_tag = 250
 
+/obj/item/clothing/head/helmet/merc
+	name = "tactical helmet"
+	desc = "A tan helmet made from advanced ceramic. Comfortable and robust."
+	icon_state = "merchelm"
+	item_flags = THICKMATERIAL | COVER_PREVENT_MANIPULATION
+	flags_inv = HIDEEARS|HIDEEYES|BLOCKHAIR
+	armor = list(melee = 65, bullet = 70, laser = 70, energy = 45, bomb = 40, bio = 0, rad = 0)
```
Comment (SpadesSlick) [inline code/modules/clothing/head/helmet.dm]: What is his even, we needed the alternative to IH armor, not SWAT Armor Tesla Bugalo. Make it have the same stats IH helmet have.

File: code/datums/uplink/grenades.dm [hunk 2]
```diff
@@ -1,45 +1,99 @@
...
 	path = /obj/item/weapon/storage/box/explosive
 
 /datum/uplink_item/item/grenades/viscerator
-	name = "viscerator grenade"
-	item_cost = 4
+	name = "Viscerator Grenade"
+	item_cost = 20
 	path = /obj/item/weapon/grenade/spawnergrenade/manhacks
 
 /datum/uplink_item/item/grenades/blob
```
Comment (SpadesSlick) [inline code/datums/uplink/grenades.dm]: Just remove thin unholy garbage from uplink.

File: code/modules/projectiles/guns/launcher/money_cannon.dm [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+	update_release_force(bling.worth)
+	if(release_force >= 1)
+		var/datum/effect/effect/system/spark_spread/s = new()
+		s.set_up(3, 1, src)
+		s.start()
+
+	return bling
+
+/obj/item/weapon/gun/launcher/money/attack_self(mob/user as mob)
+	var/disp_amount = min(input(user, "How many thaler do you want to dispense at a time? (0 to [src.receptacle_value])", "Money Cannon Settings", 20) as num, receptacle_value)
```
Comment (SpadesSlick) [inline code/modules/projectiles/guns/launcher/money_cannon.dm]: >thaler

File: code/modules/projectiles/guns/launcher/money_cannon.dm [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+	fire_sound = 'sound/weapons/guns/fire/money_launcher.ogg'
+	var/emagged = 0
+
+	var/receptacle_value = 0
+	var/dispensing = 20
+
+/obj/item/weapon/gun/launcher/money/hacked
+	emagged = 1
+
+/obj/item/weapon/gun/launcher/money/proc/vomit_cash(var/mob/vomit_onto, var/projectile_vomit)
```
Comment (SpadesSlick) [inline code/modules/projectiles/guns/launcher/money_cannon.dm]: nani the fuck

Comment (RandomColonist) [reply code/modules/projectiles/guns/launcher/money_cannon.dm]: Proc for "suiciding" with money cannon. Causes you basically shoot money in your mouth, all around and play casino win sound.

File: cev_eris.dme [hunk 5]
```diff
@@ -407,6 +407,7 @@
 #include "code\datums\underwear\underwear.dm"
 #include "code\datums\uplink\ammunition.dm"
 #include "code\datums\uplink\announcements.dm"
+#include "code\datums\uplink\armor and apparel.dm"
```
Comment (ACCount12) [inline cev_eris.dme]: No spaces in file names please. It appears to work but we are dealing with BYOND here. Better safe than weird-ass unexpected BYOND bugs down the line.

File: code/modules/mob/living/carbon/resist.dm [hunk 6]
```diff
@@ -245,7 +245,6 @@
 		buckled.user_unbuckle_mob(src)
 
 /mob/living/carbon/escape_buckle()
-	setClickCooldown(100)
```
Comment (ACCount12) [inline code/modules/mob/living/carbon/resist.dm]: what is this?

File: code/modules/modular_computers/file_system/programs/antagonist/access_decrypter.dm [hunk 7]
```diff
@@ -94,7 +94,7 @@
 	return max(operator_skill - STAT_LEVEL_ADEPT, 0) * 30
 
 /datum/computer_file/program/access_decrypter/proc/get_speed()
-	var/skill_speed_modifier = 1 + (operator_skill - STAT_LEVEL_ADEPT)/4
+	var/skill_speed_modifier = 1 + (operator_skill - STAT_LEVEL_BASIC)/3
```
Comment (ACCount12) [inline code/modules/modular_computers/file_system/programs/antagonist/access_decrypter.dm]: Another undocumented change?

File: code/modules/mob/living/carbon/resist.dm (hunk 6 above)
Comment (RandomColonist) [reply code/modules/mob/living/carbon/resist.dm]: Will document.

File: code/modules/modular_computers/file_system/programs/antagonist/access_decrypter.dm (hunk 7 above)
Comment (RandomColonist) [reply code/modules/modular_computers/file_system/programs/antagonist/access_decrypter.dm]: Will document.

**Comment by fernandos33** [comment]
meh i just want to put my vote towards what i like. some say we are mrp some say lrp some say thered n RP. fuck it idk what is it. but i like the protect mission. also would be nice to have ESPECIFIC department targets like "destroy moebius and moebius only" or "guild". thw thing is i like missions that goes towards a non valid playstyle. protect is definetly one. and i domt think taking iy straight out is nice (maybe remodeling it in some aspects. like doing the protect only when theres a "war"going on inside the ship. if you REALLY want that bad to have lots of things to shoot. but my point is. talking is nice. and fights are cool too. being in the middle of a firefight unnarmed and without armor... not so much. (its not only tjat i like the interactions. but also that i dislike mindless killing. "even tho war is fun to play. cause yiu have even odds"

**Comment by RandomColonist** [comment]
> meh i just want to put my vote towards what i like. some say we are mrp some say lrp some say thered n RP. fuck it idk what is it. but i like the protect mission. also would be nice to have ESPECIFIC department targets like "destroy moebius and moebius only" or "guild". thw thing is i like missions that goes towards a non valid playstyle. protect is definetly one. and i domt think taking iy straight out is nice (maybe remodeling it in some aspects. like doing the protect only when theres a "war"going on inside the ship. if you REALLY want that bad to have lots of things to shoot. but my point is. talking is nice. and fights are cool too. being in the middle of a firefight unnarmed and without armor... not so much. (its not only tjat i like the interactions. but also that i dislike mindless killing. "even tho war is fun to play. cause yiu have even odds"

Dude, do you really expect me to read all that crap? I won't change anything, go complain at discord or something.

**Comment by fernandos33** [comment]
> Dude, do you really expect me to read all that crap?

well. i was going to go quiet and i was for like 5 hours... but i need to say... "all that?" man you are a good coder. for real you are. and a lot of the things in this PR is really nice. but i believe i should say what i think about the PR since i would like the same said to my PRs, but hey you are calling me out for "all that", look at how much things are in your SINGLE pr this should at least be more than one pr hehehehe btw i read it all and more than once, reading is not that bad. and its not even that much tbh

**Comment by Kurgis** [comment]
I know there's a way to commit a "squash", which turns all commits into one commit. It would improve the readability of this PR greatly. Though I don't know how to perform the function myself, only that it exists.

**Comment by RandomColonist** [comment]
> I know there's a way to commit a "squash", which turns all commits into one commit. It would improve the readability of this PR greatly. Though I don't know how to perform the function myself, only that it exists.

I believe it is done on the repo side, i can't really squash myself.

File: cev_eris.dme [hunk 8]
```diff
@@ -407,17 +407,17 @@
...
 #include "code\datums\uplink\grenades.dm"
 #include "code\datums\uplink\hardsuit_modules.dm"
-#include "code\datums\uplink\highly visible and dangerous weapons.dm"
+#include "code\datums\uplink\highly_visible_and_dangerous_weapons.dm"
 #include "code\datums\uplink\implants.dm"
 #include "code\datums\uplink\medical.dm"
 #include "code\datums\uplink\neotheology.dm"
-#include "code\datums\uplink\stealth and camouflage items.dm"
-#include "code\datums\uplink\stealthy and inconspicuous weapons.dm"
+#include "code\datums\uplink\stealth and camouflage_items.dm"
```
Comment (Kurgis) [inline cev_eris.dme]: Missed this spot.

File: code/modules/clothing/head/helmet.dm [hunk 9]
```diff
@@ -58,16 +58,26 @@
 	body_parts_covered = HEAD|FACE|EARS
 	siemens_coefficient = 1
 
-/obj/item/clothing/head/helmet/tactical
-	name = "tactical helmet"
+/obj/item/clothing/head/helmet/NT
```
Comment (Kurgis) [inline code/modules/clothing/head/helmet.dm]: Seems to be an oversight here, there was a request to genericize this which was resolved, but it still remains an improper named item.

**Comment by drexample** [comment]
@RandomColonist are you still working on this?

**Comment by SpadesSlick** [comment]
I guess not.

**Pull request closed without merging**